tests/test_models.py

```diff
@@ -26,3 +30,45 @@ def test_without_conditions(model):
     gen.fit(X, C=None)
     X_gen = gen.sample(C=n)
     assert X_gen.shape == X.shape
+
+
+is_conds = [True, False]
+devices = ['cpu', 'cuda']
+@pytest.mark.parametrize("is_cond,device", list(itertools.product(is_conds, devices)))
+def test_ddpm(is_cond, device):
+    n = 100
+    X = torch.from_numpy(np.random.normal(size=(n, 5))).to(torch.float32)
+    y = torch.from_numpy(np.random.normal(size=(n, 2))).to(torch.float32)
+    len_y = y.shape[1]; len_X = X.shape[1]
+
+    backbone_args_dict = {
+        'var_dim': len_y if is_cond else len_y + len_X,
+        'cond_input_dim': len_X if is_cond else None,
+        'hid_dim': 128,
+        'num_blocks': 6,
+        'n_steps': 100,
+        'steps_dim': 128,
+        'steps_depth': 5,
+        'steps_out_dim': 8,
+        'use_cond_emb': is_cond,
+        'cond_hid_dim': None,
+        'cond_output_dim': 4 * len_X if is_cond else None,
+        'device': device,
+    }
+
+    backbone = PlainBackboneResidual(**backbone_args_dict)
+
+    betas = (1e-3, 1e-2)
+    beta_grid = 'linear'
+    sigma_method = 'beta_wave'
+    n_epochs = 100
+
+    wrapper = DDPM(backbone, betas, beta_grid, sigma_method, n_epochs=n_epochs, batch_size=100)
```

Comment:
@necroshine0, hi!
Thank you for your PR. However, I can not merge it due to the interface inconsistency. Please, look at models in the library, fix DDPM interface and create a new PR, please.